### PR TITLE
Added a telnet client that behaves like the ssh client

### PR DESCRIPTION
**Pull request by lindycoder**
The goal is to allow new drivers to use this client and not change the implementation of the that equipment's adapter.
The tests have been made to work on both client in the same suite ensuring functional parity.

### CODE REVIEW

File: netman/adapters/telnet_client.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        if isinstance(expect, basestring):
+            expect = [expect]
+        expect = [re.escape(s) for s in list(expect)]
+
+        result = self.wait_for_it(expect)
+        self.full_log += result
+
+        return result
+
+    def wait_for_it(self, expect):
```
Comment (stephanerobert) [inline netman/adapters/telnet_client.py]: Maybe call this method _wait_for(self, expect) to match the one in ssh

**Comment by lindycoder** [comment]
Comments by @stephanerobert has been addressed

File: netman/adapters/telnet_client.py (hunk 1 above)
Comment (idjaw) [inline netman/adapters/telnet_client.py]: maybe it would be good to specify more explicitly what "it" is?

**Comment by idjaw** [comment]
:+1:

**Comment by stephanerobert** [comment]
I would rename ssh in the test to make it global, other then that, Look Pretty Sweet !! (y)

**Pull request closed without merging**